### DISCUSSION TITLE
NodeRED IoT

### DISCUSSION THREAD

**Discussion by Kristofer3127** [moment/luxon#1483] [Show and tell]
Hello Luxon GitHub community! I hope all are well.

I wanted to express my gratitude for this time management tool that is Luxon! Let me share with you my project, since this is the "Show and tell" area. I have a fleet of IoT devices (microcontrollers), and I have a cloud computer (AWS EC2 Ubuntu server) that they send their sensor data to. This cloud computer uses NodeRED to receive the raw sensor data, process it, and then save it to a database (InfluxDB). To convert the unix-timestamp from the microcontrollers to the correct local time of where each IoT devices is located, I was able to get Luxon working in NodeRED! 

Luxon makes this process so much easier than programming timezone code myself, especially with the annoying day-light-savings (DST) mechanism. As such, I am curious about how Luxon keeps current with the different timezones and DST? Maybe, volunteers from around the world update Luxon every year? Again, thank you and keep up the good work.

**Comment by diesieben07** [comment]
I am happy to hear Luxon could help you with your project!

Luxon itself does not actually contain timezone or localization information, it uses the various `Intl` APIs, mainly [`Intl.DateTimeFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat). This means the "burden" of keeping that information up to date is on the platform (i.e. the browser or Node.JS) and Luxon itself can be smaller.

**Reply by Kristofer3127** [reply]
Hello @diesieben07! Thank you for the reply, just let me get clarification. If I'm not mistaken, NodeRED runs on NodeJS. So, if I understand you correctly, I need to keep NodeJS updated to keep Luxon operating properly with the most current time information, right? Furthermore, do you have an opinion of the NodeJS community? Do they do a good job at staying updated, or is this something I should worry about for long term projects?

**Reply by diesieben07** [reply]
Yes, you'll have to keep NodeJS up to date.
I am not 100% on the internals here, however I think the data actually comes from V8, which is the JavaScript engine used by NodeJS (and Chromium, for that matter). It in turn uses Unicode ICU, which is kept up to date there. So I am fairly certain you do not have to worry here, especially considering the fact that V8 / NodeJS has a reasonable interest in keeping this data up to date, so that the Intl APIs provide accurate data.

**Reply by Kristofer3127** [reply]
Hello @diesieben07! Thank you for your advice and that valuable information. My project brings many technologies together, and I can't help but feel that any one of these technologies is virtually infinitely deep. I'll never be able to learn it all, as much as I'd like to. So, once again, I am grateful for Luxon.
